library/tasks/ZarrUtils.wdl

```diff
@@ -103,7 +105,8 @@ task OptimusZarrConversion {
        --output_path_for_zarr "${sample_id}.zarr" \
        --format DirectoryStore \
        --sample_id ${sample_id} \
-       --count_matrix ${sparse_count_matrix}
+       --count_matrix ${sparse_count_matrix} \
+       --expression_data_type $expression_data_type_param
```

Comment:
You need `{}` to expand a WDL input. For all of these cases I would also use `~{}` instead of `${}`, as `~{}` is the newer form and doesn't cause conflicts with bash's `${}` expression.
```suggestion
       --expression_data_type ~{expression_data_type_param}
```

When this code was converted to WDL 1.0, the $'s should've been changed to ~'s but we must've missed it then.

Reply:
Yes, this needs fixing

Reply:
we actually have a lot of those and ticket to change them

Reply:
Actually no, I am wrong the ${} need to be changed to ~{} and we have another ticket. But $expression_data_type_param is fine because its a bash variable set in the script.

Reply:
> Actually no, I am wrong the ${} need to be changed to ~{} and we have another ticket. But $expression_data_type_param is fine because its a bash variable set in the script.

Ohhh.. yeah that is very confusing. Could you instead use an `if`? and make the bash variable all uppercase, as is the usual convention for bash variables. That would help readability for me, at least.

Reply:
I just made the changes.